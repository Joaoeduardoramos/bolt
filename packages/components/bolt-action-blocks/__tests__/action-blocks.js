import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../action-blocks.schema.yml'));
const { valign, borderless } = schema.properties;

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 60000;

describe('<bolt-action-blocks> Component', () => {
  afterAll(async () => {
    await stopTwigRenderer();
  }, timeout);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await renderTwig(
      '@bolt-components-action-blocks/action-blocks.twig',
      {
        contentItems: [
          {
            text: 'Item 1',
            url: '#!',
            icon: {
              name: 'download',
              size: 'large',
            },
          },
          {
            text: 'Item 2',
            url: '#!',
            icon: {
              name: 'copy-to-clipboard',
              size: 'large',
            },
          },
          {
            text: 'Item 3',
            url: '#!',
            icon: {
              name: 'calendar',
              size: 'large',
            },
          },
        ],
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  valign.enum.forEach(async valignChoice => {
    test(`Vertical alignment of each block's content: ${valignChoice}`, async () => {
      const results = await renderTwig(
        '@bolt-components-action-blocks/action-blocks.twig',
        {
          valign: valignChoice,
          contentItems: [
            {
              text: 'Item 1',
              url: '#!',
              icon: {
                name: 'download',
                size: 'large',
              },
            },
            {
              text:
                'Item 2: this item has more text, so it can demonstrate the vertical alignment',
              url: '#!',
              icon: {
                name: 'copy-to-clipboard',
                size: 'large',
              },
            },
            {
              text: 'Item 3',
              url: '#!',
              icon: {
                name: 'calendar',
                size: 'large',
              },
            },
          ],
        },
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  borderless.enum.forEach(async borderlessChoice => {
    test(`Border in between each block: ${borderlessChoice}`, async () => {
      const results = await renderTwig(
        '@bolt-components-action-blocks/action-blocks.twig',
        {
          borderless: borderlessChoice,
          contentItems: [
            {
              text: 'Item 1',
              url: '#!',
              icon: {
                name: 'download',
                size: 'large',
              },
            },
            {
              text: 'Item 2',
              url: '#!',
              icon: {
                name: 'copy-to-clipboard',
                size: 'large',
              },
            },
            {
              text: 'Item 3',
              url: '#!',
              icon: {
                name: 'calendar',
                size: 'large',
              },
            },
          ],
        },
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
