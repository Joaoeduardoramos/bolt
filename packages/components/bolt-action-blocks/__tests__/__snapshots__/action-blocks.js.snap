// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-action-blocks> Component Basic usage 1`] = `

<bolt-action-blocks bolt-component>
  <ul class="c-bolt-action-blocks c-bolt-action-blocks--item-max-6 c-bolt-action-blocks--vertical-align-start c-bolt-action-blocks--bordered">
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="download"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 1
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="copy-to-clipboard"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 2
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="calendar"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 3
          </div>
        </a>
      </bolt-action-block>
    </li>
  </ul>
</bolt-action-blocks>

`;

exports[`<bolt-action-blocks> Component Border in between each block: false 1`] = `

<bolt-action-blocks bolt-component>
  <ul class="c-bolt-action-blocks c-bolt-action-blocks--item-max-6 c-bolt-action-blocks--vertical-align-start c-bolt-action-blocks--borderless">
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="download"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 1
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="copy-to-clipboard"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 2
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="calendar"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 3
          </div>
        </a>
      </bolt-action-block>
    </li>
  </ul>
</bolt-action-blocks>

`;

exports[`<bolt-action-blocks> Component Border in between each block: true 1`] = `

<bolt-action-blocks bolt-component>
  <ul class="c-bolt-action-blocks c-bolt-action-blocks--item-max-6 c-bolt-action-blocks--vertical-align-start c-bolt-action-blocks--bordered">
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="download"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 1
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="copy-to-clipboard"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 2
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="calendar"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 3
          </div>
        </a>
      </bolt-action-block>
    </li>
  </ul>
</bolt-action-blocks>

`;

exports[`<bolt-action-blocks> Component Vertical alignment of each block's content: bottom 1`] = `

<bolt-action-blocks bolt-component>
  <ul class="c-bolt-action-blocks c-bolt-action-blocks--item-max-6 c-bolt-action-blocks--vertical-align-end c-bolt-action-blocks--bordered">
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="download"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 1
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="copy-to-clipboard"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 2: this item has more text, so it can demonstrate the vertical alignment
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="calendar"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 3
          </div>
        </a>
      </bolt-action-block>
    </li>
  </ul>
</bolt-action-blocks>

`;

exports[`<bolt-action-blocks> Component Vertical alignment of each block's content: center 1`] = `

<bolt-action-blocks bolt-component>
  <ul class="c-bolt-action-blocks c-bolt-action-blocks--item-max-6 c-bolt-action-blocks--vertical-align-center c-bolt-action-blocks--bordered">
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="download"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 1
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="copy-to-clipboard"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 2: this item has more text, so it can demonstrate the vertical alignment
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="calendar"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 3
          </div>
        </a>
      </bolt-action-block>
    </li>
  </ul>
</bolt-action-blocks>

`;

exports[`<bolt-action-blocks> Component Vertical alignment of each block's content: top 1`] = `

<bolt-action-blocks bolt-component>
  <ul class="c-bolt-action-blocks c-bolt-action-blocks--item-max-6 c-bolt-action-blocks--vertical-align-start c-bolt-action-blocks--bordered">
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="download"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 1
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="copy-to-clipboard"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 2: this item has more text, so it can demonstrate the vertical alignment
          </div>
        </a>
      </bolt-action-block>
    </li>
    <li class="c-bolt-action-blocks__item">
      <bolt-action-block bolt-component>
        <a href="#!"
           class="c-bolt-action-block"
        >
          <div class="c-bolt-action-block__item c-bolt-action-block__icon">
            <bolt-icon name="calendar"
                       size="large"
            >
            </bolt-icon>
          </div>
          <div class="c-bolt-action-block__item c-bolt-action-block__text">
            Item 3
          </div>
        </a>
      </bolt-action-block>
    </li>
  </ul>
</bolt-action-blocks>

`;
